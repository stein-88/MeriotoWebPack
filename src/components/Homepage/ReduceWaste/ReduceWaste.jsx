import React from 'react'
import schema from './data'
import s from './ReduceWaste.scss'

const ReduceWaste = () => {
	const { titulo, card } = schema
	return (
		<section>
			<div className="row">
				<div className="col-12">
					{titulo && <h2 className="h1 mt-5 py-4 text-muted">{titulo}</h2>}
				</div>
			</div>
			<div className="row">
				{card && card.length > 0 && card.map((ele, i) => (
					<div
						key={`reducew ${i + 1}`}
						className="col-lg-6 col-sm-12 col-md-12"
					>
						<section className={s.sectionobrigatoria}>
							{ele && ele.link
								&& (
									<a href={ele.link}>
										<section className={`position-relative ${s.seccard} ${ele.classesec || ''}`}>
											<section className={s.textobaixo}>
												<h3 className="h2 pl-4 pr-4 text-white">
													{ele.titulo && <strong>{ele.titulo}</strong>}
												</h3>
												{ele.text
													&& (
														<section className="text-right w-100">
															<span className={`mr-4 ${s.stylespan}`}>{ele.text}</span>
														</section>
													)}
											</section>
										</section>
									</a>
								)}
						</section>
					</div>
				))}
			</div>
		</section>
	)
}

export default ReduceWaste
