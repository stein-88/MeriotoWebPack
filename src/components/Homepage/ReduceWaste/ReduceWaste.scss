.sectionobrigatoria {
  padding: 1rem;
  margin: 1.5rem;

  .seccard {
    min-height: 420px;
    position: relative;
    cursor: pointer;
    border-radius: 25px;

    &.secforest {
      background: linear-gradient(0deg, #0006, #0006), url("../../../statics/images/homecard1.jpeg") center center/cover no-repeat;
    }

    &.seckid {
      background: linear-gradient(0deg, #0000004d, #0000004d), url("../../../statics/images/homecard2.jpeg") center center/cover no-repeat;
    }

    .textobaixo {
      position: absolute;
      bottom: 20px;

      .stylespan {
        background-color: #f3f3f3;
        color: #000 !important;
        cursor: pointer;
        padding: 10px 4rem;
        display: inline-block;
        text-align: center;
        outline: 0;
        border-radius: 15px;
        border-color: transparent;
        text-decoration: none !important;
      }
    }
  }
}
