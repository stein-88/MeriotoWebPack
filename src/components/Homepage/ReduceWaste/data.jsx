import s from './ReduceWaste.scss'

const schema = {
    titulo: 'Reduce Waste, Generate ESG Impact',
    card: [
        {
            link: 'https://clavo.globechain.com/corporates',
            titulo: 'Discover the benefits of listing items',
            text: 'Learn more',
            classesec: s.secforest,
        },
        {
            link: 'https://clavo.globechain.com/charities',
            titulo: 'How taking items benefits everyone',
            text: 'Learn more',
            classesec: s.seckid,
        },
    ],
}

export default schema
