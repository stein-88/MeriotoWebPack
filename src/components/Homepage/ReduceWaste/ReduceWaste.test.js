import React from 'react'
import renderer from 'react-test-renderer'
import ReduceWaste from './ReduceWaste'
import schema from './data'
const { card } = schema

describe('testing ReduceWaste', () => {
    const compReduceWaste = renderer.create(<ReduceWaste />)
    test('Should render without props', () => {
        const jsonReduceWaste = compReduceWaste.toJSON()
        expect(jsonReduceWaste).toMatchSnapshot()
    })
    test('Should have 1 title and 2 cards', () => {
        const resultCARDS = compReduceWaste.root.findAllByProps({ className: 'textobaixo' })
        const resultTitle = compReduceWaste.root.findAllByProps({ className: 'h1 mt-5 py-4 text-muted' })
        expect(resultCARDS).toHaveLength(card.length)
        expect(resultTitle).toHaveLength(1)
    })
})