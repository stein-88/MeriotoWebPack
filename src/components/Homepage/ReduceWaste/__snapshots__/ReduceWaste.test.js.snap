// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ReduceWaste Should render without props 1`] = `
<section>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <h2
        className="h1 mt-5 py-4 text-muted"
      >
        Reduce Waste, Generate ESG Impact
      </h2>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-6 col-sm-12 col-md-12"
    >
      <section
        className="sectionobrigatoria"
      >
        <a
          href="https://clavo.globechain.com/corporates"
        >
          <section
            className="position-relative seccard secforest"
          >
            <section
              className="textobaixo"
            >
              <h3
                className="h2 pl-4 pr-4 text-white"
              >
                <strong>
                  Discover the benefits of listing items
                </strong>
              </h3>
              <section
                className="text-right w-100"
              >
                <span
                  className="mr-4 stylespan"
                >
                  Learn more
                </span>
              </section>
            </section>
          </section>
        </a>
      </section>
    </div>
    <div
      className="col-lg-6 col-sm-12 col-md-12"
    >
      <section
        className="sectionobrigatoria"
      >
        <a
          href="https://clavo.globechain.com/charities"
        >
          <section
            className="position-relative seccard seckid"
          >
            <section
              className="textobaixo"
            >
              <h3
                className="h2 pl-4 pr-4 text-white"
              >
                <strong>
                  How taking items benefits everyone
                </strong>
              </h3>
              <section
                className="text-right w-100"
              >
                <span
                  className="mr-4 stylespan"
                >
                  Learn more
                </span>
              </section>
            </section>
          </section>
        </a>
      </section>
    </div>
  </div>
</section>
`;
